launcher/src/main/java/org/apache/brooklyn/launcher/BrooklynWebServer.java

```diff
@@ -447,7 +450,15 @@ public synchronized void start() throws Exception {
         rootContext.setTempDirectory(Os.mkdirs(new File(webappTempDir, "war-root")));
 
         server.setHandler(handlers);
-        server.start();
+        try {
+            server.start();
+        } catch (BindException e) {
+            // port discovery routines may take some time to clear, e.g. 250ms for SO_TIMEOUT
+            // tests fail because of this; see if adding a delay improves things
+            log.warn("Initial server start-up failed binding (retrying after a delay): "+e);
+            Time.sleep(Duration.millis(500));
+            server.start();
```

Comment:
> Why would setReuseAddress(true) improve false positives? As long as we make the same call when we start the server it should be identical? (Or are you thinking that the root cause is that the server-start call isn't using that, and that is what you'd fix?)

That's right - `setReuseAddress(true)` is used in `isPortAvailable`, but not in the actual server bind call (for web or jmx). `SO_REUSEADDR` allows you to bind to a port in `TIME_WAIT` state. So the availability check will succeed, but the server bind will fail.

I wouldn't add `setReuseAddress(true)` to the server, but will instead remove it from `isPortAvailable`.

We have parallel tests not from the same maven run, but from parallel jenkins jobs (another process).